src/sentry/digests/backends/redis.py

```diff
@@ -146,126 +94,73 @@ def validate(self):
             label='Digests',
         )
 
-    def add(self, key, record, increment_delay=None, maximum_delay=None):
+    def _get_connection(self, key):
+        return self.cluster.get_local_client_for_key(
+            '{}:t:{}'.format(self.namespace, key),
+        )
+
+    def _get_timeline_lock(self, key, duration):
+        lock_key = '{}:t:{}'.format(self.namespace, key)
+        return self.locks.get(
+            lock_key,
+            duration=duration,
+            routing_key=lock_key,
+        )
+
+    def add(self, key, record, increment_delay=None, maximum_delay=None, timestamp=None):
+        if timestamp is None:
+            timestamp = time.time()
+
         if increment_delay is None:
             increment_delay = self.increment_delay
 
         if maximum_delay is None:
             maximum_delay = self.maximum_delay
 
-        timeline_key = make_timeline_key(self.namespace, key)
-        record_key = make_record_key(timeline_key, record.key)
-
-        connection = self.cluster.get_local_client_for_key(timeline_key)
-        with connection.pipeline() as pipeline:
-            pipeline.multi()
-
-            pipeline.set(
-                record_key,
-                self.codec.encode(record.value),
-                ex=self.ttl,
+        # Redis returns "true" and "false" as "1" and "None", so we just cast
```

Comment:
This test verifies that behavior, too:
https://github.com/getsentry/sentry/blob/be5c524218471de03a64f35223d5a2b28d40b772/tests/sentry/digests/backends/test_redis.py#L16-L24